hive/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -163,6 +171,140 @@ public void renameTable(TableIdentifier from, TableIdentifier to) {
+    }
+  }
+
+  @Override
+  public boolean dropNamespace(Namespace namespace) {
+    if (namespace.levels().length != 1) {
+      throw new NoSuchNamespaceException("Namespace does not exist: %s", namespace);
+    }
+
+    try {
+      clients.run(client -> {
+        client.dropDatabase(namespace.level(0),
+            false /* deleteData */,
+            false /* ignoreUnknownDb */,
+            true /* cascade */);
+        return null;
+      });
+
+      return true;
+
+    } catch (NoSuchObjectException e) {
+      throw new NoSuchNamespaceException("Namespace %s does not exist: %s", namespace, e);
+
+    } catch (TException e) {
+      throw new RuntimeException("Failed to drop namespace " + namespace.toString() + " in Hive MataStore", e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException(
+          "Interrupted in call to drop dropDatabase(name)" + namespace.toString() + " in Hive MataStore", e);
+    }
+  }
+
+  @Override
+  public boolean setNamespaceMetadata(Namespace namespace, Map<String, String> meta) {
+    Map<String, String> currentMeta = loadNamespaceMetadata(namespace);
+    if (meta.containsKey("location")) {
+      throw new UnsupportedOperationException(
+          "Does not support change 'location' in HiveCatalog: " + namespace.toString());
+    }
+    if (meta.containsKey("name")) {
+      throw new UnsupportedOperationException(
+          "Does not support change 'name' in HiveCatalog: " + namespace.toString());
+    }
+
+    meta.forEach(currentMeta::put);
+    Database currentDb = convertToDatabase(namespace, currentMeta);
+    try {
+      clients.run(client -> {
+        client.alterDatabase(namespace.toString(), currentDb);
+        return null;
+      });
+      return true;
+    } catch (NoSuchObjectException | UnknownDBException e) {
+      throw new NoSuchNamespaceException("Namespace %s does not exist: %s", namespace.toString(), e);
+
+    } catch (TException e) {
+      throw new RuntimeException(
+          "Failed to list namespace under namespace: " + namespace.toString() + " in Hive MataStore", e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException(
+          "Interrupted in call to getDatabase(name)" + namespace.toString() + " in Hive MataStore", e);
+    }
+
+  }
+
+  @Override
+  public Map<String, String> loadNamespaceMetadata(Namespace namespace) {
+    if (namespace.levels().length != 1) {
+      throw new NoSuchNamespaceException("Namespace does not exist: %s", namespace);
+    }
+
+    try {
+      Database database = clients.run(client -> client.getDatabase(namespace.toString()));
+      return convertToMetadata(database);
+
+    } catch (NoSuchObjectException | UnknownDBException e) {
+      throw new NoSuchNamespaceException("Namespace %s does not exist: %s", namespace, e);
```

Comment:
When passing the underlying exception to `NoSuchNamespaceException`, it needs to come first. This call would embed the exception's `toString` output in the message of the thrown exception instead of wrapping it as a cause.